code/modules/chemistry/tools/pills.dm

```diff
@@ -91,6 +91,8 @@
 			qdel(src)
 			return
 		else
+			if (src.reagents.total_volume <= 0)
+				qdel(src)
```

Comment:
Wouldn't this check be more appropriate in `on_reagent_change()`?

Reply:
I had no idea that existed, yeah that makes a lot more sense. 

I'm also thinking that maybe pills should delete themselves on reagent change to any closed containers too, since it was pointed it that you can't exactly split what is probably wrapping around chems. Not sure how I would do that, though.

Reply:
to make more sense of that second paragraph, is there an `on_transfer()`? That might be more useful so pills don't qdel themselves on chem reactions if it makes sense to change it that pills qdel on being used even if not all reagents are used.